src/main/java/uk/gov/hmcts/reform/notificationservice/config/AuthServiceConfig.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.notificationservice.config;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.authorisation.ServiceAuthorisationApi;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.authorisation.validators.ServiceAuthTokenValidator;
+
+import java.util.List;
+
+@Configuration
+public class AuthServiceConfig {
+
+    @Bean
+    @ConditionalOnProperty(name = "idam.s2s-auth.url")
+    public AuthTokenValidator tokenValidator(ServiceAuthorisationApi s2sApi) {
+        return new ServiceAuthTokenValidator(s2sApi);
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "idam.s2s-auth.url", havingValue = "false")
+    public AuthTokenValidator tokenValidatorStub() {
+        return new AuthTokenValidator() {
+            @Override
+            public void validate(String token) {
+                throw new NotImplementedException();
+            }
+
+            @Override
+            public void validate(String token, List<String> roles) {
+                throw new NotImplementedException();
+            }
+
+            @Override
+            public String getServiceName(String token) {
+                return "some_service_name";
+            }
+        };
```

Comment:
is this for integration tests?

Reply:
yes, have removed for now

Reply:
i think what rafal meant is to move to integration sources rather than keeping in main

Reply:
I was wondering whether we need it for anything

Reply:
for beans to load up as auth service now will be plugged into scope. integration tests will definitely fail having s2s_url = false